haystack/preview/embedding_backends/sentence_transformers_backend.py

```diff
@@ -0,0 +1,40 @@
+from typing import List, Optional, Union, Dict
+import hashlib
+import numpy as np
+
+from haystack.lazy_imports import LazyImport
+
+with LazyImport(message="Run 'pip install farm-haystack[inference]'") as sentence_transformers_import:
+    from sentence_transformers import SentenceTransformer
```

Comment:
I know that we don't want Haystack dependencies in preview (#5569).
Maybe the [lazy_import wrapper](https://github.com/deepset-ai/haystack/blob/c26f1e94261c298ae6f500123ff23811a60100b9/haystack/lazy_imports.py) can be added in another PR.

Reply:
I'm gonna open a separate PR to port this over soon!

Reply:
https://github.com/deepset-ai/haystack/pull/5580